lib/ts/utils/dateProvider/types.ts

```diff
@@ -14,9 +14,10 @@
  */
 
 export interface DateProviderInterface {
+    getThresholdInSeconds(): number;
     now(): number;
     setClientClockSkewInMillis(clockSkewInMillis: number): void;
     getClientClockSkewInMillis(): number;
```

Comment:
there is also a recipe function called `getClockSkewInMillis`, which is kind of confusing given that these two functions also exist. From a user's point of view, which one should they call? DateProvider. getClientClockSkewInMillis(), or sessionrecipe.getClockSkewInMillis() (from a naming point of view, it's confusing).

So, i think you should come up with a better name for the recipe interface function: `getClockSkewInMillis`  or restructure how stuff is done..

Reply:
Renamed it to `calculateClockSkewInMillis`. I think that's a more appropriate name.

Reply:
Correct me if I'm wrong, but `getClockSkewInMillis` is not exposed to users from the recipe (partially because of this)

Reply:
It is not exposed but users can override it. So, I renamed it to avoid any confusion.

Reply:
Ok, just making sure.